File: app/src/main/java/com/example/barrettsbudgettracker/MainActivity.java
package com.example.barrettsbudgettracker;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private ArrayList<Transaction> transaction;
    private Adapter adapter;
    private RecyclerView recyclerView;
    private TextView balanceTextView;
    private TextView budgetTextView;
    private TextView expenseTextView;
    private FloatingActionButton addBtn;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Initialize views from XML
        recyclerView = findViewById(R.id.recyclerView);
        balanceTextView = findViewById(R.id.balance);
        budgetTextView = findViewById(R.id.budget);
        expenseTextView = findViewById(R.id.expenses);
        addBtn = findViewById(R.id.addBtn); // Initialize addBtn

        transaction = new ArrayList<>();
        transaction.add(new Transaction("Weekly budget", 500.00));
        transaction.add(new Transaction("Coffee", -20.00));
        transaction.add(new Transaction("Gas", -45.00));
        transaction.add(new Transaction("Parking", -60.00));
        transaction.add(new Transaction("Lunch", -80.00));
        transaction.add(new Transaction("Snacks", -10.00));
        transaction.add(new Transaction("Art supplies", -30.00));
        transaction.add(new Transaction("Clothes", -50.00));
        transaction.add(new Transaction("Date", -50.00));

        adapter = new Adapter(transaction);
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        updateDashboard();

        addBtn.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, TransactionActivity.class);
            startActivity(intent);
        });
    }

    private void updateDashboard() {
        double totalAmount = 0;
        double budgetAmount = 0;
        for (Transaction t : transaction) {
            totalAmount += t.getAmount();
            if (t.getAmount() > 0) {
                budgetAmount += t.getAmount();
            }
        }

        double expenseAmount = totalAmount - budgetAmount;

        balanceTextView.setText(String.format("$%.2f", totalAmount));
        budgetTextView.setText(String.format("$%.2f", budgetAmount));
        expenseTextView.setText(String.format("$%.2f", expenseAmount));
    }
}

File: app/src/main/java/com/example/barrettsbudgettracker/MainActivity.kt
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.barrettsbudgettracker.Adapter
import com.example.barrettsbudgettracker.R
import com.example.barrettsbudgettracker.Transaction
import com.example.barrettsbudgettracker.TransactionActivity
import com.google.android.material.floatingactionbutton.FloatingActionButton

class MainActivity : AppCompatActivity() {
    private lateinit var transaction: ArrayList<Transaction>
    private lateinit var adapter: Adapter
    private lateinit var recyclerView: RecyclerView
    private lateinit var balanceTextView: TextView
    private lateinit var budgetTextView: TextView
    private lateinit var expenseTextView: TextView
    private lateinit var addBtn: FloatingActionButton

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        // Initialize views from XML
        recyclerView = findViewById(R.id.recyclerView)
        balanceTextView = findViewById(R.id.balance)
        budgetTextView = findViewById(R.id.budget)
        expenseTextView = findViewById(R.id.expenses)
        addBtn = findViewById(R.id.addBtn) // Initialize addBtn

        transaction = arrayListOf(
            Transaction("Weekly budget", 500.00),
            Transaction("Coffee", -20.00),
            Transaction("Gas", -45.00),
            Transaction("Parking", -60.00),
            Transaction("Lunch", -80.00),
            Transaction("Snacks", -10.00),
            Transaction("Art supplies", -30.00),
            Transaction("Clothes", -50.00),
            Transaction("Date", -50.00)
        )

        adapter = Adapter(transaction)
        recyclerView.apply {
            this.adapter = adapter
            layoutManager = LinearLayoutManager(this@MainActivity)
        }

        updateDashboard()

        addBtn.setOnClickListener{
            val intent = Intent(this, TransactionActivity::class.java)
            startActivity(intent)
        }
    }

    private fun updateDashboard() {
        val totalAmount = transaction.map { it.amount }.sum()
        val budgetAmount = transaction.filter { it.amount > 0 }.map { it.amount }.sum()
        val expenseAmount = totalAmount - budgetAmount

        balanceTextView.text = "$%.2f".format(totalAmount)
        budgetTextView.text = "$%.2f".format(budgetAmount)
        expenseTextView.text = "$%.2f".format(expenseAmount)
    }
}

